x-pack/platform/plugins/shared/fleet/server/services/epm/elasticsearch/template/template.test.ts

```diff
@@ -55,380 +68,447 @@ describe('EPM template', () => {
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        registryElasticsearch: {
+          'index_template.data_stream': {
+            hidden: true,
+          },
+        },
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithDataStreamHidden.data_stream.hidden).toEqual(true);
+
+      const templateWithoutDataStreamHidden = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        hidden: true,
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithoutDataStreamHidden.data_stream.hidden).toEqual(true);
 
-    expect(mappings).toMatchSnapshot(path.basename(ymlPath));
-  });
+      const templateWithGlobalHiddenTrueAndDataStreamHiddenFalse = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        hidden: true,
+        registryElasticsearch: {
+          'index_template.data_stream': {
+            hidden: false,
+          },
+        },
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithGlobalHiddenTrueAndDataStreamHiddenFalse.data_stream.hidden).toEqual(true);
+
+      const templateWithoutHidden = getTemplate({
+        templateIndexPattern,
+        type: 'logs',
+        packageName: 'nginx',
+        composedOfTemplates: [],
+        templatePriority: 200,
+        isIndexModeTimeSeries: false,
+      });
+      expect(templateWithoutHidden.data_stream.hidden).toEqual(undefined);
+    });
 
-  it('tests loading system.yml', () => {
-    const ymlPath = path.join(__dirname, '../../fields/tests/system.yml');
-    const fieldsYML = readFileSync(ymlPath, 'utf-8');
-    const fields: Field[] = load(fieldsYML);
-    const processedFields = processFields(fields);
+    it('adds composed_of correctly for otel input packages of type logs', () => {
+      const composedOfTemplates = ['component1', 'component2'];
 
-    const mappings = generateMappings(processedFields);
+      const template = getTemplate({
+        templateIndexPattern: 'logs.otel-*',
+        type: 'logs',
+        packageName: 'otel-test',
+        composedOfTemplates,
+        templatePriority: 200,
+        isOtelInputType: true,
+      });
+      expect(template.composed_of).toStrictEqual([
+        OTEL_COMPONENT_TEMPLATE_MAPPINGS,
+        OTEL_COMPONENT_TEMPLATE_SETTINGS,
+        OTEL_COMPONENT_TEMPLATE_LOGS_MAPPINGS,
+        OTEL_COMPONENT_SEMCONV_RESOURCE_TO_ECS_MAPPINGS,
+        OTEL_COMPONENT_TEMPLATE_LOGS_CUSTOM_MAPPINGS,
+        STACK_COMPONENT_TEMPLATE_LOGS_MAPPINGS,
+        STACK_COMPONENT_TEMPLATE_LOGS_SETTINGS,
+        ...composedOfTemplates,
+        STACK_COMPONENT_TEMPLATE_ECS_MAPPINGS,
+        FLEET_GLOBALS_COMPONENT_TEMPLATE_NAME,
+        FLEET_AGENT_ID_VERIFY_COMPONENT_TEMPLATE_NAME,
+      ]);
```

Comment:
Could we have a test where we see also the component templates added for a package?

Reply:
It can be a unit test, it would be a test that checks the order of component templates when merging all of them, so it would include the `@package` template, the package-specific `@custom` templates and so on.

Reply:
So a test where we can have a check similar to this:
```
  expect(template.composed_of).toStrictEqual([
        STACK_COMPONENT_TEMPLATE_LOGS_MAPPINGS,
        STACK_COMPONENT_TEMPLATE_LOGS_SETTINGS,
        OTEL_COMPONENT_TEMPLATE_MAPPINGS,
        OTEL_COMPONENT_TEMPLATE_SETTINGS,
        OTEL_COMPONENT_TEMPLATE_LOGS_MAPPINGS,
        OTEL_COMPONENT_SEMCONV_RESOURCE_TO_ECS_MAPPINGS,
        "otel-test@package",   
        "logs@custom",
        OTEL_COMPONENT_TEMPLATE_LOGS_CUSTOM_MAPPINGS,
        "otel-test@custom",
        "logs-<dataset>@custom",
        STACK_COMPONENT_TEMPLATE_ECS_MAPPINGS,
        FLEET_GLOBALS_COMPONENT_TEMPLATE_NAME,
        FLEET_AGENT_ID_VERIFY_COMPONENT_TEMPLATE_NAME,
      ]);
```

Reply:
I added some new tests that checks the output of `prepareTemplate`, unfortunately it doesn't give the order as you wrote above. We do many manipulations before getting to the final list of assets.